Userland/Libraries/LibGfx/Rect.h

```diff
@@ -604,19 +604,11 @@ class Rect {
     {
         if (constrain_rect.contains(*this))
             return *this;
-        T move_x = 0, move_y = 0;
-        if (right() > constrain_rect.right())
-            move_x = constrain_rect.right() - right();
-        if (bottom() > constrain_rect.bottom())
-            move_y = constrain_rect.bottom() - bottom();
-        if (x() < constrain_rect.x())
-            move_x = x() - constrain_rect.x();
-        if (y() < constrain_rect.y())
-            move_y = y() - constrain_rect.y();
-        auto rect = *this;
-        if (move_x != 0 || move_y != 0)
-            rect.translate_by(move_x, move_y);
-        return rect;
+        Point<T> top_left_point { top_left() };
+        Point<T> bottom_right_point { x() + width(), y() + height() };
+        top_left_point.constrain(constrain_rect);
+        bottom_right_point.constrain(constrain_rect);
```

Comment:
If the top left point falls outside of `constrain_rect`, but the bottom right falls inside of it, and the rect's size is small enough that it fits entirely inside of `constrain_rect`, we are now resizing it which seems incorrect.

I think the logic should be:
1) Try to move rect inside the constrain rect, aligning the top left points where necessary
2) Resize the rect to be smaller if it doesn't fit inside the constrain rect

Maybe we should have some tests for this as well :^)

Reply:
That makes sense. I think I've misunderstood the intent of this function. For what I'm doing, I don't want the rect to move if it's partially outside the input rect's bounds, so I've introduced a new function `Rect::clipped_to()`, which does the same thing as I've implemented here. I'm open to suggestions for a better name though.

Reply:
I feel as though this new `::clipped_to` is very similar to what `::intersect` does?

Reply:
`::intersected` works fine here. I've removed it.